mcp-spi/src/main/java/io/modelcontextprotocol/logger/McpLogger.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2025-2025 the original author or authors.
+ */
+
+package io.modelcontextprotocol.logger;
+
+/**
+ * @author Aliaksei Darafeyeu
+ */
+public interface McpLogger {
```

Comment:
This logging interface doesn't seem ideal; it feels like all logging levels should support attaching a `Throwable` cause, the number of methods feels excessive, and there's no way to lazily evaluate string parameters if the message is omitted.  Maybe a single `void log(McpSchema.LoggingLevel level, Supplier<String> messageSupplier, @Nullable Throwable cause)` method would make more sense.